pytorch_lightning/core/optimizer.py

```diff
@@ -249,78 +252,79 @@ def _configure_optimizers(
-                    rank_zero_warn(
-                        f"The lr scheduler dict contains the key(s) {keys_to_warn}, but the keys will be ignored."
-                        " You need to call `lr_scheduler.step()` manually in manual optimization.",
-                        category=RuntimeWarning,
-                    )
-
-                scheduler = {key: scheduler[key] for key in scheduler if key not in invalid_keys}
-                lr_schedulers.append({**default_config, **scheduler})
-            else:
-                lr_schedulers.append({**default_config, "scheduler": scheduler})
-        else:
-            if isinstance(scheduler, dict):
-                # check provided keys
-                extra_keys = [k for k in scheduler.keys() if k not in default_config.keys()]
-                if extra_keys:
-                    rank_zero_warn(
-                        f"Found unsupported keys in the lr scheduler dict: {extra_keys}", category=RuntimeWarning
-                    )
-                if "scheduler" not in scheduler:
-                    raise MisconfigurationException(
-                        'The lr scheduler dict must have the key "scheduler" with its item being an lr scheduler'
-                    )
-                if "interval" in scheduler and scheduler["interval"] not in ("step", "epoch"):
-                    raise MisconfigurationException(
-                        'The "interval" key in lr scheduler dict must be "step" or "epoch"'
-                        f' but is "{scheduler["interval"]}"'
-                    )
-                scheduler["reduce_on_plateau"] = isinstance(
-                    scheduler["scheduler"], optim.lr_scheduler.ReduceLROnPlateau
+        if isinstance(scheduler, dict):
+            # check provided keys
+            extra_keys = [k for k in scheduler.keys() if k not in default_config.keys()]
+            if extra_keys:
+                rank_zero_warn(
+                    f"Found unsupported keys in the lr scheduler dict: {extra_keys}", category=RuntimeWarning
+                )
+            if "scheduler" not in scheduler:
+                raise MisconfigurationException(
+                    'The lr scheduler dict must have the key "scheduler" with its item being an lr scheduler'
+                )
+            if "interval" in scheduler and scheduler["interval"] not in ("step", "epoch"):
+                raise MisconfigurationException(
+                    'The "interval" key in lr scheduler dict must be "step" or "epoch"'
+                    f' but is "{scheduler["interval"]}"'
+                )
+            scheduler["reduce_on_plateau"] = isinstance(scheduler["scheduler"], optim.lr_scheduler.ReduceLROnPlateau)
+            if scheduler["reduce_on_plateau"] and scheduler.get("monitor", None) is None:
+                raise MisconfigurationException(
+                    "The lr scheduler dict must include a monitor when a `ReduceLROnPlateau` scheduler is used."
```

Comment:
```suggestion
                    "The LR scheduler `dict` must include a monitor when a `ReduceLROnPlateau` scheduler is used."
```

Reply:
moved code can't be changed, out of scope.